src/sentry/models/event.py

```diff
@@ -320,75 +310,310 @@ def site(self):
+        directly from Snuba data.
+        """
+        if name in ['user']:
+            from sentry.interfaces.user import User
+            # This is a fake version of the User interface constructed
+            # from just the data we have in Snuba.
+            snuba_user = {
+                'id': self.user_id,
+                'email': self.email,
+                'username': self.username,
+                'ip_address': self.ip_address,
+            }
+            if any(v is not None for v in snuba_user.values()):
+                return User.to_python(snuba_user)
+        return self.interfaces.get(name)
+
+    def get_event_type(self):
+        return self.__dict__.get('type', 'default')
+
+    # These should all have been normalized to the correct values on
+    # the way in to snuba, so we should be able to just use them as is.
+    @property
+    def ip_address(self):
+        return self.__dict__['ip_address']
+
+    @property
+    def title(self):
+        return self.__dict__['title']
+
+    @property
+    def culprit(self):
+        return self.__dict__['culprit']
+
+    @property
+    def location(self):
+        return self.__dict__['location']
+
+    # ============================================
+    # Snuba implementations of django Fields
+    # ============================================
+    @property
+    def datetime(self):
+        """
+        Reconstruct the datetime of this event from the snuba timestamp
+        """
+        # dateutil seems to use tzlocal() instead of UTC even though the string
+        # ends with '+00:00', so just replace the TZ with UTC because we know
+        # all timestamps from snuba are UTC.
+        return parse_date(self.timestamp).replace(tzinfo=pytz.utc)
+
+    @property
+    def time_spent(self):
+        return None
+
+    @property
+    def id(self):
+        # Because a snuba event will never have a django row id, just return
+        # the hex event_id here. We should be moving to a world where we never
+        # have to reference the row id anyway.
+        return self.event_id
+
+    def next_event(self, environments=[]):
+        from sentry.utils import snuba
+
+        conditions = [['event_id', '!=', self.id]]
+
+        if len(environments) > 0:
+            conditions.append(['environment', 'IN', environments])
+
+        result = snuba.raw_query(
+            start=self.datetime,  # gte current event
+            end=datetime.utcnow(),  # will be clamped to project retention
+            selected_columns=self.selected_columns,
+            conditions=conditions,
+            filter_keys={
+                'project_id': [self.project_id],
+                'issue': [self.group_id],
+            },
+            orderby=['timestamp', 'event_id'],
+            limit=5
+        )
+
+        if 'error' in result or len(result['data']) == 0:
+            return None
+
+        events = [e for e in result['data'] if
+                  e['timestamp'] == self.timestamp and
```

Comment:
We can just pass these conditions to Snuba and get 1 event. The only reason that we were selecting 5 events and post-filtering them in python before was that Postgres couldn't handle this condition well (too slow) but in snuba we should be able to just specify the condition in the query and get the 1 matching event.

Reply:
Ah right, updated to drop this and select 1